f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -413,13 +415,25 @@ def _get_ports_on_network(self, context, network_id=None):
         )
 
     @log_helpers.log_method_call
-    def _get_l7policies(self, context, listeners):
+    def _get_l7policies(self, context, loadbalancer, listeners):
         """Get l7 policies filtered by listeners."""
         l7policies = []
         if listeners:
             listener_ids = [l['id'] for l in listeners]
-            policies = self.plugin.db.get_l7policies(
-                context, filters={'listener_id': listener_ids})
+
+            def get_db_policies():
+                if cfg.CONF.f5_driver_perf_mode in (1, 3):
+                    db_policies = []
+                    for l1 in loadbalancer.listeners:
```

Comment:
Maybe we could have better names for variable and mode name which gives a clear meaning, temporarily, I could accept it.